wren-ui/src/apollo/server/services/askingService.ts

```diff
@@ -302,6 +345,92 @@ export class AskingService implements IAskingService {
       wrenAIAdaptor,
       threadResponseRepository,
     });
+    this.threadRecommendQuestionBackgroundTracker =
+      new ThreadRecommendQuestionBackgroundTracker({
+        telemetry,
+        wrenAIAdaptor,
+        threadRepository,
+      });
+
+    this.mdlService = mdlService;
+  }
+
+  public async getThreadRecommendationQuestions(
+    threadId: number,
+  ): Promise<ThreadRecommendQuestionResult> {
+    const thread = await this.threadRepository.findOneBy({ id: threadId });
+    if (!thread) {
+      throw new Error(`Thread ${threadId} not found`);
+    }
+
+    // handle not started
+    const res: ThreadRecommendQuestionResult = {
+      status: RecommendQuestionResultStatus.NOT_STARTED,
+      questions: [],
+      error: null,
+    };
+    if (thread.queryId && thread.questionsStatus) {
+      res.status = RecommendQuestionResultStatus[thread.questionsStatus]
+        ? RecommendQuestionResultStatus[thread.questionsStatus]
+        : res.status;
+      res.questions = thread.questions || [];
+      res.error = thread.questionsError as WrenAIError;
+    }
+    return res;
+  }
+
+  public async generateThreadRecommendationQuestions(
+    threadId: number,
+  ): Promise<void> {
+    const thread = await this.threadRepository.findOneBy({ id: threadId });
+    if (!thread) {
+      throw new Error(`Thread ${threadId} not found`);
+    }
+
+    const tasks = this.threadRecommendQuestionBackgroundTracker.getTasks();
+    const taskKey =
+      this.threadRecommendQuestionBackgroundTracker.taskKey(thread);
+    if (tasks[taskKey]) {
+      logger.debug(
+        `thread "${threadId}" recommended questions are generating, skip the current request`,
+      );
+      return;
+    }
```

Comment:
nit: you could implement a method like: `getTask` or `ifExists` to check the existence of task with thread id.